Tasks/DotNetCoreInstallerV0/externals/get-os-distro.sh

```diff
@@ -156,12 +156,38 @@ get_legacy_os_name() {
     return 1
 }
 
-primaryName=$(get_current_os_name || echo "")
-legacyName=$(get_legacy_os_name || echo "")
+get_machine_architecture() {
+    eval $invocation
+
+    if command -v uname > /dev/null; then
+        CPUName=$(uname -m)
+        case $CPUName in
+        armv7l)
+            echo "arm"
+            return 0
+            ;;
+        aarch64)
+            echo "arm64"
+            return 0
+            ;;
+        esac
+    fi
+
+    # Always default to 'x64'
+    echo "x64"
+    return 0
+}
+
+osName=$(get_current_os_name || echo "")
+legacyOsName=$(get_legacy_os_name || echo "")
+arch=$(get_machine_architecture || echo "")
+
+primaryName="$osName-$arch"
+legacyName="$legacyOsName-$arch"
```

Comment:
why legacy needs to append arch now. Was that a bug earlier?

Reply:
No, this was bug in the PR which I have updated now.